appinventor/components-ios/src/extensions/posenet.swift

```diff
@@ -0,0 +1,255 @@
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import Foundation
+import WebKit
+
+
+open class PosenetExtension: NonvisibleComponent, WKScriptMessageHandler {
+  fileprivate final let _ERROR_WEBVIEWER_NOT_SET: String =
+  "You must specify a WebViewer using the WebViewer designer property before you can call"
+  fileprivate final let _ERROR_JSON_PARSE_FAILED: Int = 101
+  fileprivate final let _MODEL_URL: String =
+  "https://storage.googleapis.com/tfjs-models/savedmodel/posenet/mobilenet/quant2/050/"
+  fileprivate final let _BACK_CAMERA: String = "Back"
+  fileprivate final let _FRONT_CAMERA: String = "Front"
+  
+  fileprivate var _webview: WKWebView?  // Double check ! syntax
+  private var _webviewerApiSource: String = ""
+  private var _webviewerApi: WKUserScript? // double check ! syntax
+  
+  fileprivate var _keyPoints: [String: [String]] = [:]
+  fileprivate var _minPoseConfidence: Double = 0.1
+  fileprivate var _minPartConfidence: Double = 0.5
+  fileprivate lazy var _cameraMode: String = _FRONT_CAMERA
+  fileprivate var _initialized: Bool = false
+  fileprivate var _enabled: Bool = true
+  fileprivate var _backgroundImage: String = ""
+  
+  private enum IllegalStateError: Error {
+    case webviewerNotSet
+  }
+  
+  
+  public override init(_ parent: ComponentContainer) {
+    super.init(parent)
+    _keyPoints["nose"] = []
+    _keyPoints["leftEye"] = []
+    _keyPoints["rightEye"] = []
+    _keyPoints["leftEar"] = []
+    _keyPoints["rightEar"] = []
+    _keyPoints["leftShoulder"] = []
+    _keyPoints["rightShoulder"] = []
+    _keyPoints["leftElbow"] = []
+    _keyPoints["rightElbow"] = []
+    _keyPoints["leftWrist"] = []
+    _keyPoints["rightWrist"] = []
+    _keyPoints["leftHip"] = []
+    _keyPoints["rightHip"] = []
+    _keyPoints["leftKnee"] = []
+    _keyPoints["rightKnee"] = []
+    _keyPoints["leftAnkle"] = []
+    _keyPoints["rightAnkle"] = []
+  }
+  
+  /**
+   Configure the current webview. Add a WKScriptMessageHandler
+   to handle the following Javascript calls necessary for
+   Posenet Extension:
+          ready()
+          reportImage(String dataURL)
+          reportResult(String result)
+          error(int errorCode, String errorMessage)
+   Equivalent to the @Javascript interface in the Android Posenet Extension.
+   */
+  private func configureWebView(_ webview: WKWebView) {
+    self._webview = webview
+    let config = webview.configuration
+    // config.defaultWebpagePreferences.allowsContentJavaScript = true
+    let controller = config.userContentController
+    controller.add(self, name: "ready")
+    controller.add(self, name: "reportImage")
+    controller.add(self, name: "reportResult")
+    controller.add(self, name: "error")
+  }
+  
+  
+  // MARK: PoseNet Properties
+  
+  
+//  @objc open var WebViewer: WebViewer {
+//    get {
+//      return _webviewer
+//    }
+//    set {
+//      if 
+//    }
+//  }
+//  
```

Comment:
The code you showed me earlier today had a more substantial implementation regarding setting the WebViewer property. I'm not sure what happened here, but did you forget to commit some code?